paasta_tools/vitesscluster_tools.py

```diff
@@ -1,27 +1,544 @@
+TABLET_TYPES = ["primary", "migration"]
+WEB_PORT = "15000"
+GRPC_PORT = "15999"
+
+
+# Environment variables
+VTCTLD_EXTRA_ENV = [
+    {
+        "name": "WEB_PORT",
+        "value": WEB_PORT,
+    },
+    {
+        "name": "GRPC_PORT",
+        "value": GRPC_PORT,
+    },
+    {
+        "name": "TOPOLOGY_FLAGS",
+        "value": "",
+    },
+]
+
+VTTABLET_EXTRA_ENV = [
+    {
+        "name": "SHARD",
+        "value": "0",
+    },
+    {
+        "name": "EXTERNAL_DB",
+        "value": "1",
+    },
+    {
+        "name": "ROLE",
+        "value": "rdonly",
+    },
+    {
+        "name": "WEB_PORT",
+        "value": WEB_PORT,
+    },
+    {
+        "name": "GRPC_PORT",
+        "value": GRPC_PORT,
+    },
+    {
+        "name": "VAULT_ROLEID",
+        "valueFrom": {
+            "secretKeyRef": {
+                "name": "paasta-vitessclusters-secret-vitess-k8s-vault-vttablet-approle-roleid",
+                "key": "vault-vttablet-approle-roleid",
+            }
+        },
+    },
+    {
+        "name": "VAULT_SECRETID",
+        "valueFrom": {
+            "secretKeyRef": {
+                "name": "paasta-vitessclusters-secret-vitess-k8s-vault-vttablet-approle-secretid",
+                "key": "vault-vttablet-approle-secretid",
+            }
+        },
+    },
+]
+
+# Vault auth related variables
+VTGATE_EXTRA_ENV = [
+    {
+        "name": "VAULT_ROLEID",
+        "valueFrom": {
+            "secretKeyRef": {
+                "name": "paasta-vitessclusters-secret-vitess-k8s-vault-vtgate-approle-roleid",
+                "key": "vault-vtgate-approle-roleid",
+            }
+        },
+    },
+    {
+        "name": "VAULT_SECRETID",
+        "valueFrom": {
+            "secretKeyRef": {
+                "name": "paasta-vitessclusters-secret-vitess-k8s-vault-vtgate-approle-secretid",
+                "key": "vault-vtgate-approle-secretid",
+            }
+        },
+    },
+]
+
+
+# Extra Flags
+VTADMIN_EXTRA_FLAGS = {"grpc-allow-reflection": "true"}
+
+VTCTLD_EXTRA_FLAGS = {
+    "disable_active_reparents": "true",
+    "security_policy": "read-only",
+}
+
+VTTABLET_EXTRA_FLAGS = {
+    "log_err_stacks": "true",
+    "grpc_max_message_size": "134217728",
+    "init_tablet_type": "replica",
+    "queryserver-config-schema-reload-time": "1800",
+    "dba_pool_size": "4",
+    "vreplication_heartbeat_update_interval": "60",
+    "vreplication_tablet_type": "REPLICA",
+    "keep_logs": "72h",
+    "enable-lag-throttler": "true",
+    "throttle_check_as_check_self": "true",
+    "db_charset": "utf8mb4",
+    "disable_active_reparents": "true",
+}
+
+
+def get_updated_environment_variables(original_list: List[Dict], update_dict: Dict):
+    for env in original_list:
+        if env["name"] in update_dict:
+            env["value"] = update_dict[env["name"]]
+            del update_dict[env["name"]]
+    original_list.extend([{"name": k, "value": v} for k, v in update_dict.items()])
+    return original_list
```

Comment:
what are your thoughts on storing kv dicts for env vars and having a `better_name_here_to_k8s()` function that took a dict and turned it into a list of dicts? would save us some complexity in copying/modifying dicts :)